wherehows-web/app/services/banners.ts

```diff
@@ -0,0 +1,87 @@
+import Service from '@ember/service';
+import { computed, get } from '@ember/object';
+import { NotificationEvent } from 'wherehows-web/services/notifications';
+import { delay } from 'wherehows-web/utils/promise-delay';
+
+/**
+ * Expected properties to be found on a basic banner object added to our list
+ */
+export interface IBanner {
+  content: string;
+  type: NotificationEvent;
+  isExiting: boolean;
+  isDismissable: boolean;
+  iconName: string;
+}
+
+/**
+ * When creating a new banner, helps to determine whether that banner isDismissable
+ */
+const isDismissableMap: { [key: string]: boolean } = {
+  [NotificationEvent['info']]: true,
+  [NotificationEvent['confirm']]: true,
+  [NotificationEvent['error']]: false
+};
+
+/**
+ * When creating a new banner, helps to determine the kind of font awesome icon indicator will be on
+ * the left side of the banner
+ */
+const iconNameMap: { [key: string]: string } = {
+  [NotificationEvent['info']]: 'info-circle',
+  [NotificationEvent['confirm']]: 'exclamation-circle',
+  [NotificationEvent['error']]: 'times-circle'
+};
+
+export default class BannerService extends Service {
+  /**
+   * Our cached list of banner objects to be rendered
+   * @type {Array<IBanner>}
+   */
+  banners: Array<IBanner> = [];
+
+  /**
+   * Computes the banners list elements to toggle the isShowingBanners flag, which not only activates the
+   * banner component but also triggers the navbar and app-container body to shift up/down to accommodate
+   * for the banner component space
+   * @type {ComputedProperty<boolean>}
+   */
+  isShowingBanners = computed('banners.@each.isExiting', function() {
+    const banners = get(this, 'banners');
+    // Note: If we have no banners, flag should always be false. If we have more than one banner, flag
+    // should always be true, BUT if we only have one banner and it is in an exiting state we can go ahead
+    // and trigger this to be false to line up with the animation
+    return banners.length > 0 && (banners.length > 1 || !banners[0].isExiting);
+  });
+
+  /**
+   * Method to actually take care of removing the first banner from our queue.
+   */
+  async dequeue(): Promise<void> {
+    // Note: Since dequeuing the banner will remove it from the DOM, we don't want to actually dequeue
+    // until the removal animation, which takes 0.7 seconds, is completed.
+    const animationSpeed = 0.8;
```

Comment:
Is there a better way to have the same reference passed into the stylesheet animation? I think having a delay value that's separate from the value used in the spreadsheet will be a maintenance issue because there's no way to for them to be in sync? Let me know if that makes sense.
You can do with a data-attribute binding on the element, that you can probably ready from the stylesheet

Reply:
Good point! Will look into that.